tiledb/sm/query/readers/result_tile.cc

```diff
@@ -173,7 +173,7 @@ void ResultTile::erase_tile(const std::string& name) {
 
   // Handle attribute tile
   for (auto& at : attr_tiles_) {
-    if (at.second.has_value() && at.first == name) {
+    if (at.first == name) {
```

Comment:
Please add a comment about why `at.second.has_value()` is always true here. We still have the check in the clause above.

Reply:
This is actually removing a change made in @shaunrd0 's last PR while debugging that accidentally made it in. Calling reset on a non-set optional is a no-op so the check had no effect.